src/trace/trace.cpp

```diff
@@ -159,7 +162,16 @@ void Trace::begin_record()
 
 void Trace::end_record()
 {
-    stopping_time_ = time::now();
+    // write missing program_end events
+    auto now = lo2s::time::now();
+    for (pid_t pid : missing_program_end_event_cache_)
+    {
+        Log::debug() << "Missed program_begin event for process " << pid
+                     << ", assuming it exited successfully";
+        archive_(thread_sample_locations_.at(pid)) << otf2::event::program_end{ now, EXIT_SUCCESS };
```

Comment:
Check if this can be moved into the monitor, which holds the referenced location